### PR TITLE
Allow anyOf semantics in addition to allOf for policies. Fixes GH-41

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: controller/persistence/base_entity.go [hunk 1]
```diff
@@ -44,12 +44,18 @@ const (
...
 	FieldRoleAttributes = "roleAttributes"
 	FieldCreatedAt      = "createdAt"
 	FieldUpdatedAt      = "updatedAt"
 	FieldTags           = "tags"
+
+	SemanticAllOf = "AllOf"
+	SemanticAnyOf = "AnyOf"
 )
 
+var validSemantics = []string{SemanticAllOf, SemanticAnyOf}
```
Comment (andrewpmartinez) [inline controller/persistence/base_entity.go]: Seems weird this is here, but it doesn't hurt anything.

File: controller/persistence/edge_router_policy_store_test.go [hunk 2]
```diff
@@ -51,6 +51,10 @@ func (ctx *TestContext) testEdgeRouterPolicyInvalidValues(_ *testing.T) {
 	err := ctx.create(policy)
 	ctx.EqualError(err, fmt.Sprintf("the value '[%v]' for 'identityRoles' is invalid: no identities found with the given names/ids", invalidId))
 
+	policy.IdentityRoles = []string{AllRole, roleRef("other")}
+	err = ctx.create(policy)
+	ctx.EqualError(err, fmt.Sprintf("the value '[%v %v]' for 'identityRoles' is invalid: if using %v, it should be the only role specified", AllRole, roleRef("other"), AllRole))
```
Comment (andrewpmartinez) [inline controller/persistence/edge_router_policy_store_test.go]: As a heads up `t *Testing.T` is unique per `t.Run()` -  each `t.Run()` is run in its own go routine and the `t` has go routine sensitive state. `ctx` fields that use `require.Assertions` from `require.New(t)` should not be shared between different levels of `t.Run()`. Doing so will cause a panic error that hides the test results diffs/error messages/etc.

It isn't pretty but for each `t.Run()` if `require` is going to be use I would expect to see a new `require.New(t)`.

File: controller/persistence/base_entity.go (hunk 1 above)
Comment (plorenz) [reply controller/persistence/base_entity.go]: It's shared across multiple entity types, so I put it there b/c I couldn't think of a better place.

File: controller/persistence/edge_router_policy_store_test.go (hunk 2 above)
Comment (plorenz) [reply controller/persistence/edge_router_policy_store_test.go]: Adding the following in a follow up PR

```
func (ctx *TestContext) testContextChanged(t *testing.T) {
       ctx.req = require.New(t)
}
```

**Pull request merged**